counter/src/counterSlice.ts

```diff
@@ -72,7 +72,7 @@ export const counterSlice = createSlice({
 
 // RootStateInterface is defined as including at least this slice and any other slices that
 // might be added by a calling package
-type RootStateInterface = { counter: CounterState } & Record<string, any>;
+type RootStateInterface = { counter: CounterState };
```

Comment:
I wonder if `RootStateInterface` and `AppDispatchInterface` are correct any more now that we've narrowed them.  That's what the "main-app" will be passing in and that works from a typing perspective (because contravariance) but from within "counter" maybe these are just `StateInterface` and `DispatchInterface`?

Reply:
updated.